echopype/consolidate/split_beam_angle.py

```diff
@@ -0,0 +1,574 @@
+"""
+Contains functions necessary to compute the split-beam (alongship/athwartship)
+angles and add them to a Dataset.
+"""
+from typing import List, Optional, Tuple
+
+import numpy as np
+import xarray as xr
+
+from ..echodata import EchoData
+
+
+def _get_splitbeam_angle_power_CW(ds_beam: xr.Dataset) -> Tuple[xr.Dataset, xr.Dataset]:
+    """
+    Obtains the split-beam angle data from power encoded data with CW waveform.
+
+    Parameters
+    ----------
+    ds_beam: xr.Dataset
+        An ``EchoData`` beam group containing angle information needed for
+        split-beam angle calculation
+
+    Returns
+    -------
+    theta_fc: xr.Dataset
+        The calculated split-beam alongship angle
+    phi_fc: xr.Dataset
+        The calculated split-beam athwartship angle
+
+    Raises
+    ------
+    NotImplementedError
+        If all ``beam_type`` values are not equal to 1
+
+    Notes
+    -----
+    Can be used on both EK60 and EK80 data
+
+    Computation done for ``beam_type=1``:
+    ``physical_angle = ((raw_angle * 180 / 128) / sensitivity) - offset``
+    """
+
+    # raw_angle scaling constant
+    conversion_const = 180.0 / 128.0
+
+    # TODO: is this function useful or just annoying? Should we put it outside this function?
+    def compute_split_beam_beamtype1(angle_type: str) -> xr.Dataset:
+        """Compute a split-beam angle for ``beam_type=1`` based on ``angle_type``"""
+        return (
+            conversion_const
+            * ds_beam[f"angle_{angle_type}"]
+            / ds_beam[f"angle_sensitivity_{angle_type}"]
+            - ds_beam[f"angle_offset_{angle_type}"]
+        )
+
+    # ensure that the beam_type is appropriate for calculation
+    if np.all(ds_beam["beam_type"].data == 1):
+
+        # obtain split-beam alongship angle
+        theta_fc = compute_split_beam_beamtype1(angle_type="alongship")
+
+        # obtain split-beam athwartship angle
+        phi_fc = compute_split_beam_beamtype1(angle_type="athwartship")
```

Comment:
Some data like the OOI data files have some transducers that are split-beam and some are single beam, and in those cases we know we pad the raw (electric) `angle_alongship` and `angle_athwartship` with NaN but these variables have the same dimension as the Sv data variable.

I think we should allow `add_splitbeam_angle` to work on this type of data (so have to change the `np.all` check), but issue a warning that some channels are not split-beam so that users are aware.

The computation itself in `compute_split_beam_beamtype1` should just work with NaNs so it's a matter of allowing it and issue the warning.

Reply:
As we have discussed, I replaced `np.all(ds_beam["beam_type"].data == 1)` with `np.all(ds_beam["beam_type"].data != 0)`.